File: Data Structures & Algorithms/maximum-subarray/submission-0.cs
public class Solution {
    public int MaxSubArray(int[] nums) {

        int n = nums.Length;
        int result = nums[0];

        for (int i = 0; i < n; i++) //durchlaufe ganzes array einmal
        {
            int current = 0; //Speiche arraysumme des derzeitgen subarrays

            for (int j = i; j < n; j++) //verkleinere das Array jedes Mal um alle möglichen SUbarray abzubilden
            {
                current += nums[j]; //berechne die Summe dieses Subarrays
                result = Math.Max(result,current); //speicher linksläufige subarray ergeebnisse und bilde ale restlichen ergebnisse ab
            }

        }
        return result;
    }
}

File: Data Structures & Algorithms/sliding-window-maximum/submission-0.py
class Solution:
    def maxSlidingWindow(self, nums: List[int], k: int) -> List[int]:
        res = []
        val = 0
        for i in range(len(nums)-k+1):
            for j in range(i,i+k):
                val = max(nums[j],val)
            res.append(val)
            val = 0 
        return res
